packages/sd-jwt/src/action-handler.ts

```diff
@@ -70,28 +69,43 @@ export class SDJwtPlugin implements IAgentPlugin {
 
   /**
    * Get the key to sign the SD-JWT
-   * @param issuer - did url like did:exmaple.com#key-1
+   * @param identifier - identifier like a did and other forms of identifiers
    * @param context - agent instance
    * @returns the key to sign the SD-JWT
    */
-  private async getSignKey(issuer: string, context: IRequiredContext) {
-    debug(`Getting signing key for issuer ${issuer}`)
-    const identifier = await context.agent.didManagerGet({
-      did: issuer.split('#')[0],
-    })
-    const doc = await mapIdentifierKeysToDocWithJwkSupport({ identifier, vmRelationship: 'assertionMethod' }, context)
-    if (!doc || doc.length === 0) {
-      throw new Error('No key found for signing')
-    }
-    const key = doc.find((key) => key.meta.verificationMethod.id === issuer)
-    if (!key) {
-      throw new Error(`No key found with the given id: ${issuer}`)
+  async getSignKey (
+      identifier: string,
+      context: IRequiredContext
+      //todo make the type of key known
+  ): Promise<{ alg: string; key: any }> {
+    if (identifier.startsWith('did:')) {
+      const didDocument = await context.agent.resolveDid({ didUrl: identifier });
+      const key = didDocument.verificationMethod?.find(vm => vm.publicKeyJwk);
+      if (!key) {
+        throw new Error(`No verification key found for DID: ${identifier}`);
+      }
+      const alg = key.publicKeyJwk.alg;
+      return { alg, key };
+    } else if (identifier.includes('x5c')) {
+      const x5cKey = await context.agent.getCertificateChain({ identifier });
+      if (!x5cKey || !Array.isArray(x5cKey)) {
+        throw new Error(`Invalid x5c header found for identifier: ${identifier}`);
+      }
+      //fixme: change this value to the real alg value
+      const alg = 'RS256';
+      return { alg, key: { x5c: x5cKey } };
+    } else {
+      const key = await context.agent.keyManagerGet({ kid: identifier });
+      const alg = key.meta?.alg;
```

Comment:
This is too naive. The kid could be either a Veramo keyRef and thus either an alias or publicKeyHex. Or it could be (and more likely) an actual kid in the form of a jwkThumbprint. The kmsLocal implementation stores thumbPrints in the key metadata. Probably we need to do that for the new Musap KMS as well. For now you can test against the KML local implementation

Reply:
added functionality for fetching by jwkThumbprints in the crypto-extensions pr and tests for that